include/aikido/perception/DetectedObject.hpp

```diff
@@ -0,0 +1,89 @@
+#ifndef AIKIDO_PERCEPTION_DETECTEDOBJECT_HPP_
+#define AIKIDO_PERCEPTION_DETECTEDOBJECT_HPP_
+
+#include <string>
+#include "aikido/io/yaml.hpp"
+
+namespace aikido {
+namespace perception {
+
+/// DetectedObject delegates a detected object from a third party
+/// perception algorithm.
+
+/// A perception algorithm should send information for an object via ROS
+/// visualization_msgs/Marker message like the following:
+/// Marker.header.frame_id -> detectionFrameID
+/// Marker.ns + "_" + Marker.id -> objUid (identity in DART world)
+/// Marker.ns -> objAssetDBKey (determines visual asset, see ObjectDatabase)
+/// Marker.text -> yamlStr
+
+/// yamlStr contains any types of additional
+/// information of the object in YAML format.
+/// It also can overwrite the Asset Database Key
+/// using "db_key". (see ObjectDatabase for details)
+/// Here is an example:
+/// \code
+/// {
+///   "db_key": "food_item",
+///   "score": 0.9,
+///   "success_rates": [0.4, 0.9, 0.1, 0.2],
+///   "strategies": ["vertical_0", "vertical_90", "tilted_0", "tilted_90"],
+/// }
+/// \endcode
+
+/// Clarification: objUid = DART world ID; objDbKey = visual asset key in ObjectDatabase.
+
+class DetectedObject
+{
+public:
+  /// Construct a \c DetectedObject
+  /// \param[in] objUid Unique ID for object in DART world. Same UID -> Same Object
+  /// \param[in] objAssetDBKey Key for ObjectDatabase passed into constructor of PoseEstimatorModule. Defines visuals / assets.
```

Comment:
Maybe `assetKey`?

Reply:
Yep, was trying to avoid confusion with the existing class "ObjectDatabase", but honestly, I think renaming it to "AssetDatabase" makes the most sense.